code/game/objects/items/devices/radio/detpack.dm

```diff
@@ -0,0 +1,323 @@
+	. = ..()
+	if(istype(W, /obj/item/device/multitool))
+		if(user.mind && user.mind.cm_skills && user.mind.cm_skills.engineer < SKILL_ENGINEER_METAL)
+			user.visible_message("<span class='notice'>[user] fumbles around figuring out how to use the [src].</span>",
+			"<span class='notice'>You fumble around figuring out how to use [src].</span>")
+			var/fumbling_time = 30
+			if(!do_after(user, fumbling_time, TRUE, 5, BUSY_ICON_BUILD))
+				return
+			if(prob((SKILL_ENGINEER_METAL - user.mind.cm_skills.engineer) * 20))
+				to_chat(user, "<font color='danger'>After several seconds of your clumsy meddling the [src] buzzes angrily as if offended. You have a <b>very</b> bad feeling about this.</font>")
+				timer = 0 //Oops. Now you fucked up. Immediate detonation.
+		user.visible_message("<span class='notice'>[user] begins disarming [src] with [W].</span>",
+		"<span class='notice'>You begin disarming [src] with [W].</span>")
+		if(do_after(user, 30, TRUE, 5, BUSY_ICON_BUILD))
+			user.visible_message("<span class='notice'>[user] disarms [src].</span>",
+			"<span class='notice'>You disarm [src].</span>")
+			armed = 0
+			update_icon()
+
+/obj/item/device/radio/detpack/attack_hand(mob/user as mob)
+	if(armed)
+		to_chat(user, "<font color='warning'>Active anchor bolts are holding it in place! Disarm [src] first to remove it!</font>")
+		return
+	if(plant_target)
+		user.visible_message("<span class='notice'>[user] begins unsecuring [src] from [plant_target].</span>",
+		"<span class='notice'>You begin unsecuring [src] from [plant_target].</span>")
+		if(!do_after(user, 30, TRUE, 5, BUSY_ICON_BUILD))
+			return
+		user.visible_message("<span class='notice'>[user] unsecures [src] from [plant_target].</span>",
+		"<span class='notice'>You unsecure [src] from [plant_target].</span>")
+		nullvars()
+		update_icon()
+	return ..()
+
+/obj/item/device/radio/detpack/proc/nullvars()
+	if(istype(plant_target, /atom/movable) && plant_target.loc)
+		var/atom/movable/T = plant_target
+		//to_chat(world, "<font color='red'>DEBUG: Drag delay reset. Target drag delay: [T.drag_delay], Stored drag delay: [target_drag_delay]</font>")
+		if(T.drag_delay == 3)
+			T.drag_delay = target_drag_delay //reset the drag delay of whatever we attached the detpack to
+		T.vis_contents -= src
+	plant_target = null //null everything out now
+	target_drag_delay = null
+	armed = FALSE
+	update_icon()
+
+/obj/item/device/radio/detpack/receive_signal(datum/signal/signal)
+	if(signal?.encryption != code || !on)
+		return
+
+	if(!armed)
+		if(!plant_target) //has to be planted on something to begin detonating.
+			return
+		armed = TRUE
+		//bombtick()
+		processing_timers.Add(src)
+		update_icon()
+		//to_chat(world, "<font color='red'>DEBUG: Detpack Triggered: [loc]</font>")
+	else
+		armed = FALSE
+		//to_chat(world, "<font color='red'>DEBUG: Detpack Disarmed: [loc]</font>")
+		update_icon()
+
+	if(master && wires & 1)

```

Comment:
Never use raw values with bit flags, you must always have a define. Even if this has existed before now, add a define now that you're touching it.